src/ocm/components/clusterWizard/HostDiscovery.tsx

```diff
@@ -67,7 +69,12 @@ const HostDiscovery = ({ cluster }: { cluster: Cluster }) => {
     clearAlerts();
 
     const params: V2ClusterUpdateParams = {};
-    HostDiscoveryService.setPlatform(params, values.usePlatformIntegration);
+    HostDiscoveryService.setPlatform(
+      params,
+      values.usePlatformIntegration,
+      isClusterPlatformTypeVsphere(cluster),
```

Comment:
Here you need to know which is the enabled platform integration, not the platformType which is the thing being set.